bin/inx-chronicle/src/api/config.rs

```diff
@@ -1,29 +1,96 @@
 // Copyright 2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+use std::time::Duration;
+
 use derive_more::From;
+use regex::RegexSet;
 use serde::{Deserialize, Serialize};
 use tower_http::cors::AllowOrigin;
 
-use super::error::ConfigError;
+use super::{error::ConfigError, SecretKey};
 
 /// API configuration
 #[derive(Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
 #[serde(default)]
 pub struct ApiConfig {
     pub port: u16,
     pub allow_origins: Option<SingleOrMultiple<String>>,
+    pub password_hash: String,
+    pub password_salt: String,
+    #[serde(with = "humantime_serde")]
+    pub jwt_expiration: Duration,
+    pub public_routes: Vec<String>,
+    pub identity_path: Option<String>,
 }
 
 impl Default for ApiConfig {
     fn default() -> Self {
         Self {
-            port: 9092,
-            allow_origins: Some(String::from("*").into()),
+            port: 8042,
+            allow_origins: Some("*".to_string().into()),
+            password_hash: "c42cf2be3a442a29d8cd827a27099b0c".to_string(),
+            password_salt: "saltines".to_string(),
+            // 72 hours
+            jwt_expiration: Duration::from_secs(72 * 60 * 60),
+            public_routes: Default::default(),
+            identity_path: None,
         }
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct ApiData {
+    pub port: u16,
+    pub allow_origins: Option<SingleOrMultiple<String>>,
```

Comment:
Do we still need the `Option<>` here? I can't really think of a use-case right now.